File: reverse.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    string s;
    cin >> s;

    string s1 = "";
    for (int i = 0; i < s.length(); i++) {
        s1 = s[i] + s1;
    }
    cout << "Method 1 : " << s1 << endl;

    string s2 = "";
    s2 = s;
    reverse(s2.begin(), s2.end());
    cout << "Method 2 : " << s2 << endl;

    cout << "Original String : " << s << endl;
    
    return 0;
}

File: reverse.java
import java.util.Scanner;

public class reverse
{
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);

        String S = s.nextLine();
        
        String S1 = "";
        for (int i = 0; i < S.length(); i++) {
            S1 = S.charAt(i) + S1;
        }
        System.out.println("Method 1 : " + S1);

        String S2 = new String(S);
        String rev = new StringBuffer(S2).reverse().toString();
        System.out.println("Method 2 : " + rev);

        System.out.println("Original String : " + S);
        
        s.close();
    }
}